draft-irtf-cfrg-vdaf.md

```diff
@@ -3977,13 +3977,41 @@ parameter with the same level has been used with the same input share before.
 The whole preparation phase MUST NOT be run more than once for a given
 combination of input share and level.
 
+<!-- TODO: Text about counts -->
+
 ~~~
 def is_valid(agg_param, previous_agg_params):
-    (level, _) = agg_param
-    return all(
-        level != other_level
-        for (other_level, _) in previous_agg_params
-    )
+    # Exit early if this is the first time.
+    if len(previous_agg_params) < 1:
+      return True
+
+    (level, prefixes, threshold, prefix_counts) = agg_param
+    (last_level, last_prefixes, last_threshold, last_prefix_counts) = previous_agg_params[-1]
+
+    # Check that prefix_count for prefixes are above threshold.
+    for count in prefix_counts:
+      if count < threshold:
+        return False
+
+    # Check that level increased by one, and the threshold is the same
+    if level != last_level + 1:
+      return False
+    if threshold != last_threshold:
+      return False
+
+    # Check that prefixes are suffixes of the last level's prefixes,
+    # and that the counts of siblings sum up to at most the parent's count.
```

Comment:
This code is hard to follow. Let's try to encapsulate some of the bit fiddling details with high level funtions, like `is_parent_of(parent, child)`.

Reply:
Added a helper `get_prefix` that contains the bit shift.